src/table/table.component.tsx

```diff
@@ -65,10 +76,54 @@ const Table = React.memo((props: TableProps): React.ReactElement => {
     totalDataCount,
   ]);
 
-  const tableInstance = useTable({ columns: displayedColumns, data });
+  const defaultColumn = React.useMemo(
+    () => ({
+      minWidth: 30,
+      width: 150 + additionalHeaderSpace,
+    }),
+    []
+  );
+
+  // displayedColumns should be left as a record of exactly WHAT data is being passed to the table instance
+  // As such, it is best to not use or amend it otherwise
+  // Instead, refer to visibleColumns for a record of what is currently being displayed
+  // visibleColumns also contains exact details of how columns are being displayed, e.g. width, isResizing, etc.
+  const tableInstance = useTable(
+    {
+      columns: displayedColumns,
+      data,
+      defaultColumn,
+    },
+    useResizeColumns,
+    useFlexLayout,
+    useColumnOrder
+  );
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    setColumnOrder,
+    visibleColumns,
+  } = tableInstance;
+
+  const updater = (result: any): string[] => {
+    // Unlike displayedColumns, visibleColumns retains the current column order
+    const items = Array.from(visibleColumns);
+    const [reorderedItem] = items.splice(result.source.index, 1);
+    items.splice(result.destination.index, 0, reorderedItem);
+
+    return items.map((column: Column) => {
+      return column.Header?.toString() ?? '';
+    });
+  };
 
-  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
-    tableInstance;
+  const handleOnDragEnd = (result: DropResult): void => {
+    if (!result.destination) return;
+    setColumnOrder(updater(result));
+  };
```

Comment:
Again, loss of codecov here. Not even sure how to trigger this behaviour in a unit test. Worst comes to worst we can override the codecov threshold and just merge in?